I'm not sure if this will be ported upstream, but it's working for me, so I
guess a patch is acceptable.

cc https://github.com/kovidgoyal/kitty/issues/10304

diff --git i/kitty/cell_vertex.glsl w/kitty/cell_vertex.glsl
index cabed1495..e1df40e75 100644
--- i/kitty/cell_vertex.glsl
+++ w/kitty/cell_vertex.glsl
@@ -362,6 +362,7 @@ void main() {
     bg_alpha = if_one_then(is_special_cell, 1.f, bg_alpha);
     // Selection and cursor
     bg_alpha = if_one_then(cell_data.has_block_cursor, effective_cursor_opacity, bg_alpha);
+    bg_alpha = mix(bg_alpha, bg_alpha * inactive_text_alpha, (1.f - cell_has_default_bg) * (1.f - is_special_cell));
     bg = if_one_then(float(is_selected & BIT_MASK), if_one_then(use_cell_for_selection_bg, color_to_vec(fg_as_uint), color_to_vec(highlight_bg)), bg);
     vec3 background_rgb = if_one_then(cell_data.has_block_cursor, mix(bg, cell_data.cursor.bg, cursor_opacity), bg);
     background = background_rgb;
